backend/games_api.py

```diff
@@ -0,0 +1,70 @@
+import os
+import requests
+import time
+from datetime import datetime, timedelta, timezone
+from dotenv import load_dotenv
+from pyrate_limiter import Duration, Rate, Limiter, InMemoryBucket
+
+# Load environment variables from the .env file
+load_dotenv()
+
+# API details
+API_URL = "https://apiv2.legiontd2.com/games"
+API_KEY = os.getenv("API_KEY")
+MAX_LIMIT = 50  # Adjusted based on the API's maximum limit of 50
+
+# Set up rate limiting
+rate = Rate(5, Duration.SECOND)
+burst_rate = Rate(100, Duration.MINUTE)
+limiter = Limiter(InMemoryBucket([rate]))
+burst_limiter = Limiter(InMemoryBucket([burst_rate]))
```

Comment:
We should move this into a data class that contains all the configuration. On startup we can create a new instance of the class and provide it to any function that needs a value from it. See this blog on data classes: https://www.dataquest.io/blog/how-to-use-python-data-classes/